dask_expr/shuffle.py

```diff
@@ -653,6 +672,23 @@ def _simplify_down(self):
 
 
 class SetPartition(SetIndex):
+    """Shuffles the DataFrame according to its new divisions.
+
+    Simplifies the Expression to blockwise pre-processing, shuffle and
+    blockwise post-processing expressions.
```

Comment:
Nothing to do here, just a thought: "should we have multiple expressions for this, or just one expression and a more complex `_layer` method?"

The benefit of one expression is that it's probably simpler to reason about and maintain.

The benefit of multiple expressions is that it probably optimizes better (blockwises can be fused, for example)

Reply:
We will reuse SetPartition for sort values, that’s why I made 2